Library/Homebrew/cask/audit.rb

```diff
@@ -282,6 +284,66 @@ def check_token_conflicts
       add_warning "possible duplicate, cask token conflicts with Homebrew core formula: #{core_formula_url}"
     end
 
+    def check_token_valid
+      return unless @strict
+
+      add_warning "cask token is not lowercase" if cask.token.downcase!
+
+      add_warning "cask token contains non-ascii characters" unless cask.token.ascii_only?
+
+      add_warning "cask token + should be replaced by -plus-" if cask.token.include? "+"
+
+      add_warning "cask token @ should be replaced by -at-" if cask.token.include? "@"
+
+      add_warning "cask token whitespace should be replaced by hyphens" if cask.token.include? " "
+
+      add_warning "cask token underscores should be replaced by hyphens" if cask.token.include? "_"
+
+      if cask.token.match?(/[^a-z0-9\-]/)
+        add_warning "cask token should only contain alphanumeric characters and hyphens"
+      end
+
+      add_warning "cask token should not contain double hyphens" if cask.token.include? "--"
+
+      return unless cask.token.end_with?("-") || cask.token.start_with?("-")
+
+      add_warning "cask token should not have leading or trailing hyphens"
+    end
+
+    def check_token_bad_words
+      return unless @strict
+
+      add_warning "cask token contains .app" if cask.token.end_with? ".app"
+
+      if cask.token.end_with? "alpha", "beta", "release candidate"
+        add_warning "cask token contains version designation"
+      end
+
+      add_warning "cask token mentions launcher" if cask.token.end_with? "launcher"
```

Comment:
The token reference mentions:

> Inevitably, there are a small number of exceptions not covered by the rules. Don’t hesitate to contact the maintainers if you have a problem.

This one has a few, where `Launcher` is an intrinsic part of the app’s name and would be nonsensical without it ([SpaceLauncher](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/spacelauncher.rb), [DelayedLauncher](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/delayedlauncher.rb)). To not leave users wondering how to proceed on those cases, I suggest one of two options:

1. If *any* warning from this check would fire, also add a short message informing the user they should still submit the non-compliant name and we can discuss it in the PR.
2. Make this check behind a flag, that doesn’t run on normal `audit` but does run on CI.

I think **1** is more useful in this case. But we could also go some third way.

Reply:
> If any warning from this check would fire, also add a short message informing the user they should still submit the non-compliant name and we can discuss it in the PR.

I think this is something that should be documented in the PR template (for example: "Warnings can be discussed, Errors should be resolved"). 

> Make this check behind a flag, that doesn’t run on normal audit but does run on CI.

It is only running on strict audits (`--strict` or `--new-cask`)